src/QuickCheckVEngine/Main.hs

```diff
@@ -184,25 +184,26 @@ main = withSocketsDo $ do
   addrB <- resolve (impBIP flags) (impBPort flags)
   socA <- open "implementation-A" addrA
   socB <- open "implementation-B" addrB
-  sendDIIPacket socA diiEnd
-  _ <- recvRVFITrace socA False
-  sendDIIPacket socB diiEnd
-  _ <- recvRVFITrace socB False
+  socATraceVer <- rvfiNegotiateVersion socA "implementation A" (optVerbosity flags)
+  socBTraceVer <- rvfiNegotiateVersion socB "implementation B" (optVerbosity flags)
+  let connectionA = (socA, socATraceVer, "implementation A", (optVerbosity flags))
```

Comment:
This should be a proper data type rather than a tuple.

Reply:
Just pushed a version with a data type, but got a `Data constructor not in scope:` error so I also added a helper function to create it... Probably just a beginners mistake.